quotes_recommender/user_store/user_store_redis.py

```diff
@@ -59,6 +62,47 @@ def scan(
         hits = self._client.scan(cursor=cursor, match=search_str, count=batch_size, _type=type_filter)
         return [hit.decode(TXT_ENCODING) for hit in hits[1]]
 
+    def get_all_users(
+        self,
+        search_str: Optional[str] = None,
+        batch_size: int = DEFAULT_BATCH_SIZE,
+    ) -> dict[str, list[int]]:
+        """
+        Use SSCAN function to receive key-value results for a search string.
+        :param search_str: Search string.
+        :param batch_size: Number of returned results.
+        :return: Dictionary mapping each key to a list of its members
+
+        References:
+            - https://redis-py.readthedocs.io/en/stable/#redis.Redis.scan_iter
+        """
+        users_data = {}
+        for key in self._client.scan_iter(match=search_str, count=batch_size):
+            key_str = key.decode(TXT_ENCODING)
+            members = [int(member.decode(TXT_ENCODING)) for member in self._client.smembers(key_str)]
+            users_data[key_str] = members
+
+        return users_data
```

Comment:
Soll diese Funktion auch außerhalb von der Klasse aufgerufen werden? Wenn nicht, dann gerne in `_get_all_users` (leading underscore) umbenennen.

Reply:
Ist der eigene User damit ausgeschlossen? Es kann ja sein, dass der eingeloggte user mit zurückgegeben wird bei der `scan_iter` function. Den brauchen wir ja nicht, oder?

Reply:
Demnach hätte nämlich der eingeloggte user mit sich selbst die immer größte Set-Similarity, vor allem wenn du die Funktion in `get_most_similar_user` aufrufst/als Hilfsfunktion verwendest.

Reply:
Ja das hatte ich im Hinterkopf und wolltes es umsetzen habe aber dann beim Proggen dann wieder vergessen. Gutes Auge :)